pkg/node/store.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package node
+
+import (
+	"path"
+	"time"
+
+	"github.com/cilium/cilium/pkg/kvstore"
+	"github.com/cilium/cilium/pkg/kvstore/store"
+)
+
+var (
+	// nodesStorePrefix is kvstore prefix for the shared store for node
+	// resources
+	//
+	// WARNING - STABLE API: Do not change this
+	nodeStorePrefix = path.Join(kvstore.BaseKeyPrefix, "state", "nodes", "v1")
+
+	nodeStore *store.SharedStore
+)
+
+func (n *Node) GetKeyName() string {
```

Comment:
This might be better named `GetKeyPath` (or just `GetNodeKey` but this is also ok (My confusion arose because I Thought this was the name of the key, the value of which we look up and then lookup the actual value in the kvstore.